lib/modelMaterialsCommon.js

```diff
@@ -727,6 +729,85 @@ function lightDefaults(gltf) {
     }
 }
 
+function getShaderVariable(accessor) {
+  var type = accessor.type;
+  if (type === 'SCALAR') {
+    return 'float';
+  }
+  return type.toLowerCase();
+}
+
+function ensureSemanticExistenceForPrimitive(gltf, primitive) {
+  var accessors = gltf.accessors;
+  var materials = gltf.materials;
+  var techniques = gltf.techniques;
+  var programs = gltf.programs;
+  var shaders = gltf.shaders;
+
+  var attributes = primitive.attributes;
+  var materialId = primitive.material;
+  var material = materials[materialId];
+  var techniqueId;
+  var technique;
+  if (defined(material)) {
+    techniqueId = material.technique;
+    technique = techniques[techniqueId];
+  }
+  var programId;
+  var program;
+  if (defined(technique)) {
+    programId = technique.program;
+    program = programs[programId];
+  }
+  var vertexShaderId;
+  var vertexShader;
+  if (defined(program)) {
+    vertexShaderId = program.vertexShader;
+    vertexShader = shaders[vertexShaderId];
+  }
```

Comment:
Won't a lot of these checks always be true since it has gone through the modelMaterialsCommon? It should always have a material/technique/program. This also lets you simplify the loop.